pkg/cluster/operation/destroy.go

```diff
@@ -368,3 +407,191 @@ func DestroyComponent(getter ExecutorGetter, instances []spec.Instance, cls spec
 
 	return nil
 }
+
+// DestroyTombstone remove the tombstone node in spec and destroy them.
+// If returNodesOnly is true, it will only return the node id that can be destroy.
+func DestroyTombstone(
+	getter ExecutorGetter,
+	cluster *spec.Specification,
+	returNodesOnly bool,
+	options Options,
+	tlsCfg *tls.Config,
+) (nodes []string, err error) {
+	return DestroyClusterTombstone(getter, cluster, returNodesOnly, options, tlsCfg)
+}
+
+// DestroyClusterTombstone remove the tombstone node in spec and destroy them.
+// If returNodesOnly is true, it will only return the node id that can be destroy.
+func DestroyClusterTombstone(
+	getter ExecutorGetter,
+	cluster *spec.Specification,
+	returNodesOnly bool,
+	options Options,
+	tlsCfg *tls.Config,
+) (nodes []string, err error) {
+	instCount := map[string]int{}
+	for _, component := range cluster.ComponentsByStartOrder() {
+		for _, instance := range component.Instances() {
+			instCount[instance.GetHost()] = instCount[instance.GetHost()] + 1
+		}
+	}
+
+	var pdClient = api.NewPDClient(cluster.GetPDList(), 10*time.Second, tlsCfg)
+
+	binlogClient, err := api.NewBinlogClient(cluster.GetPDList(), tlsCfg)
+	if err != nil {
+		return nil, errors.AddStack(err)
+	}
+
+	filterID := func(instance []spec.Instance, id string) (res []spec.Instance) {
+		for _, ins := range instance {
+			if ins.ID() == id {
+				res = append(res, ins)
+			}
+		}
+		return
+	}
+
+	maybeDestroyMonitor := func(comp spec.Component, instances []spec.Instance, id string) error {
```

Comment:
And if we remove the arg `compName` from `StopAndDestroyInstance`, the `comp` arg for `maybeDestroyMonitor` is not needed